Gems/AWSCore/Code/Source/Credential/AWSDefaultCredentialHandler.cpp

```diff
@@ -88,6 +115,14 @@ namespace AWSCore
             SetProfileCredentialsProvider(Aws::MakeShared<Aws::Auth::ProfileConfigFileAWSCredentialsProvider>(
                 AWSDEFAULTCREDENTIALHANDLER_ALLOC_TAG, m_profileName.c_str()));
         }
+
+        bool allowAWSMetadata = false;
+        AWSCoreInternalRequestBus::BroadcastResult(allowAWSMetadata, &AWSCoreInternalRequests::IsAllowedAWSMetadataQueries);
+        if (allowAWSMetadata)
```

Comment:
IMHO IsAllowedMetadataQueries != go fetch AWS creds from an EC2 instance. I know its one of the many metadata things that can be fetched, but name is slightly opaque for what is a pretty serious security control.

I would assume IsAllowedAWSMetadataQueries is the uber (don't fetch config and region etc) control and theres more of a specific control. Unless this is the only use for setting.

Reply:
discussed offline, landed on `AllowAWSMetadataCredentials` as user-facing setting name so it's clearer what the scope of impact